### PR TITLE
Upgrade wdio to v8

### PR DESCRIPTION
**Pull request by longfeiwAmazonDE**
_Description of changes:_
1. Upgrade wdio to v8
2. Upgrade aws-sdk to v3
3. Bump version to v8.0.0-rc
4. Update project to ESM
5. Emit both CJS require and ESM import

_Testings:_
1. Ran the example test:
```
npm run example

> wdio-aws-device-farm-service@8.0.0-rc example
> npx wdio ./example/wdio.conf.js
Execution of 1 workers started at 2023-05-24T08:24:59.996Z

[0-0] RUNNING in chrome - file:///example/devicefarm.test.js
[0-0] PASSED in chrome - file:///example/devicefarm.test.js

 "spec" Reporter:
------------------------------------------------------------------
[chrome 113.0.5672.127 windows #0-0] Running: chrome (v113.0.5672.127) on windows
[chrome 113.0.5672.127 windows #0-0] Session ID: 68235ef7daadab653c3c17e5f8e7388a
[chrome 113.0.5672.127 windows #0-0]
[chrome 113.0.5672.127 windows #0-0] » /example/devicefarm.test.js
[chrome 113.0.5672.127 windows #0-0] webdriver.io page
[chrome 113.0.5672.127 windows #0-0]    ✓ should have the right title
[chrome 113.0.5672.127 windows #0-0]
[chrome 113.0.5672.127 windows #0-0] 1 passing (8.7s)


Spec Files:      1 passed, 1 total (100% completed) in 00:00:55
```

2. Ran the lint command:
```
npm run lint

> wdio-aws-device-farm-service@8.0.0-rc lint
> eslint -c .eslintrc.cjs --ext .ts .
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -49,21 +71,22 @@
     "aws-sdk": "^2.991.0"
```
Comment (arnulfojr) [inline package.json:71]: I wonder if we should also upgrade to use AWS SDK v3 within the major version bump 🤔

**Review by arnulfojr — APPROVED** [review]

File: tsconfig.json [hunk 2]
```diff
@@ -1,13 +1,16 @@
...
   "compilerOptions": {
     "declaration": true,
     "declarationMap": true,
+    "declarationDir": "types",
+    "emitDeclarationOnly": true,
     "resolveJsonModule": true,
     "removeComments": true,
     "strictFunctionTypes": false,
-    "lib": ["dom"],
+    "lib": ["dom", "ES2020"],
```
Comment (arnulfojr) [inline tsconfig.json:11]: Why we use ES2020?

the tsconfig from node16 is using ES2021

https://github.com/tsconfig/bases/blob/main/bases/node16.json#L6

Comment (longfeiwAmazonDE) [reply tsconfig.json]: Might be a typo. Fixed.

**Review by arnulfojr — COMMENTED** [review]
LGTM but tests are failing 💔

**Comment by longfeiwAmazonDE** [comment]
> LGTM but tests are failing 💔

Thanks for helping me with the unit tests 💐!

**Pull request merged**